substrate/frame/migrations/src/lib.rs

```diff
@@ -756,10 +792,20 @@ impl<T: Config> Pallet<T> {
 	}
 
 	/// Fail the current runtime upgrade, caused by `migration`.
+	///
+	/// When the `try-runtime` feature is enabled, this function will panic.
+	// Allow unreachable code so it can compile without warnings when `try-runtime` is enabled.
+	#[allow(unreachable_code)]
```

Comment:
Could be that it requires this to be `cfg_attr`, depending on whether clippy forbids unused lints.

Reply:
There is a clippy lint for unused clippy lints. So if you put the `unreachable_code` without there being any unreachable code, then it would error.  
But we will see, maybe it does not trigger 🤷‍♂️

Reply:
Seems like no issue with current CI

Reply:
It wasn't erroring, but I've added cfg_attr anyways in case we change how we use clippy